redux/react/ConnectMacro.hx

```diff
@@ -220,17 +316,16 @@ class ConnectMacro
 		});
 	}
 
-	static function updateCtor(fields:Array<Field>)
+	static function updateCtor(fields:Array<Field>, mapStateIsPartial:Bool)
 	{
 		var propsArg = { name: 'props', type: macro :Dynamic };
 		var contextArg = { name: 'context', type: macro :Dynamic };
 
-		var updateStateExpr = macro {
-			__state = mapState(context.store.getState(), props);
+		var initCacheExpr = macro __state = mapState(context.store.getState(), props);
 
-			if (__state != null)
-				state = react.ReactUtil.copy(state == null ? {} : state, __state);
```

Comment:
However, states are _not_ merged if mapState returns a "complete" state. Should they be merged?